usl_models/tests/atmo_ml/datasets_test.py

```diff
@@ -1,194 +1,181 @@
-        bucket_name=bucket_name,
-        spatiotemporal_file_names=spatiotemporal_file_names,
-        spatial_file_name=spatial_file_name,
-        lu_index_file_name=lu_index_file_name,
-        batch_size=batch_size,
-        time_steps_per_day=time_steps_per_day,
-        storage_client=mock_storage_client(),
-    )
-
-    # Check that the generator yields correctly batched data
-    for i, inputs in enumerate(dataset_generator):
-        spatiotemporal_batch = inputs["spatiotemporal"]
-        spatial_batch = inputs["spatial"]
-        lu_index_batch = inputs["lu_index"]
-
-        assert spatiotemporal_batch.shape == (
-            batch_size,
-            100,
-            100,
-            1,
-        ), "Spatiotemporal shape mismatch"
-        assert spatial_batch.shape == (100, 100, 1), "Spatial data shape mismatch"
-        assert lu_index_batch.shape == (100 * 100,), "LU index shape mismatch"
-
-        print(f"Batch {i + 1} spatiotemporal shape: {spatiotemporal_batch.shape}")
-        print(f"Batch {i + 1} spatial shape: {spatial_batch.shape}")
-        print(f"Batch {i + 1} LU index shape: {lu_index_batch.shape}")
+import io
+from unittest.mock import MagicMock
+from usl_models.atmo_ml.datasets import create_atmo_dataset, load_labels_from_cloud
+
+
+class TestAtmoMLDataset(unittest.TestCase):
+    @mock.patch("google.cloud.storage.Client")
+    @mock.patch("google.cloud.firestore.Client")
+    def test_create_atmo_dataset_structure(
+        self, mock_firestore_client, mock_storage_client
+    ):
+        """Test creating AtmoML dataset from GCS with expected structure and shapes."""
+        # Set up mock bucket and blobs
+        mock_storage_client_instance = mock_storage_client.return_value
+        mock_bucket = MagicMock()
+        mock_storage_client_instance.bucket.return_value = mock_bucket
+
+        # Define mock URLs for each data type
+        mock_spatial_url = "gs://test-data-bucket/spatial/mock_spatial_data.npy"
+        mock_spatiotemporal_urls = [
+            f"gs://test-data-bucket/spatiotemporal/mock_spatiotemporal_data_{i}.npy"
+            for i in range(3)
+        ]
+        mock_lu_index_url = "gs://test-data-bucket/lu_index/mock_lu_index_data.npy"
+        mock_label_urls = [
+            f"gs://test-label-bucket/labels/mock_label_data_{i}.npy" for i in range(3)
+        ]
+
+        # Configure Firestore to return these URLs
+        mock_firestore_client.collection.return_value.document.return_value.get.return_value.to_dict.side_effect = [  # noqa: E501
+            {"url": mock_spatial_url},
+            {"url": mock_spatiotemporal_urls[0]},
+            {"url": mock_spatiotemporal_urls[1]},
+            {"url": mock_spatiotemporal_urls[2]},
+            {"url": mock_lu_index_url},
+            {"url": mock_label_urls[0]},
+            {"url": mock_label_urls[1]},
+            {"url": mock_label_urls[2]},
+        ]
+
+        # Simulate spatial data blob
+        mock_spatial_blob = MagicMock()

```

Comment:
Let's move this into a util function to avoid code duplication. Something like:
def mock_numpy_blob(array: np.ndarray) -> storage.Blob: ...